apps/aechannel/src/aesc_set_delegates_tx.erl

```diff
@@ -0,0 +1,276 @@
+            , {nonce                 , Nonce}]) ->
+    channel = aeser_id:specialize_type(ChannelId),
+    account = aeser_id:specialize_type(FromId),
+    lists:all(fun(Id) -> account =:= aeser_id:specialize_type(Id) end,
+              IDelegates ++ RDelegates),
+    #channel_set_delegates_tx{channel_id             = ChannelId,
+                              from_id                = FromId,
+                              initiator_delegate_ids = IDelegates,
+                              responder_delegate_ids = RDelegates,
+                              payload                = Payload,
+                              state_hash             = StateHash,
+                              round                  = Round,
+                              ttl                    = TTL,
+                              fee                    = Fee,
+                              nonce                  = Nonce}.
+
+-spec for_client(tx()) -> map().
+for_client(#channel_set_delegates_tx{channel_id             = ChannelId,
+                                     from_id                = FromId,
+                                     initiator_delegate_ids = IDelegates,
+                                     responder_delegate_ids = RDelegates,
+                                     payload                = Payload,
+                                     state_hash             = StateHash,
+                                     round                  = Round,
+                                     ttl                    = TTL,
+                                     fee                    = Fee,
+                                     nonce                  = Nonce}) ->
+    #{<<"channel_id">>             => aeser_api_encoder:encode(id_hash, ChannelId),
+      <<"from_id">>                => aeser_api_encoder:encode(id_hash, FromId),
+      <<"initiator_delegate_ids">> => [aeser_api_encoder:encode(id_hash, Id)
+                                       || Id <- IDelegates],
+      <<"responder_delegate_ids">> => [aeser_api_encoder:encode(id_hash, Id)
+                                       || Id <- RDelegates],
+      <<"payload">>                => aeser_api_encoder:encode(transaction, Payload),
+      <<"state_hash">>             => aeser_api_encoder:encode(state, StateHash),
+      <<"round">>                  => Round,
+      <<"ttl">>                    => TTL,
+      <<"fee">>                    => Fee,
+      <<"nonce">>                  => Nonce}.
+
+serialization_template(?INITIAL_VSN) ->
+    [ {channel_id               , id}
+    , {from_id                  , id}
+    , {initiator_delegate_ids   , [id]}
+    , {responder_delegate_ids   , [id]}
+    , {payload                  , binary}
+    , {state_hash               , binary}
+    , {round                    , int}
+    , {ttl                      , int}
+    , {fee                      , int}
+    , {nonce                    , int}
+    ].
+
+-spec version(tx()) -> non_neg_integer().
```

Comment:
If this function were moved to the top of the "behaviour API" section, it would be closer to the macro definition.